p4-16/spec/P4-16-draft-spec.mdk

```diff
@@ -4876,22 +4877,18 @@ match_kind {
 ~ End P4Example
 
 These identifiers correspond to the P4~14~ match kinds with the same
-names. The semantics of these annotations is actually irrelevant for
-describing the behavior of the P4 abstract machine; how they are used
-influences only the control-plane API and the actual implementation of
-the look-up table. From the point of view of the P4 program, a look-up
-table is an abstract finite map that is given a key and produces as a
-result either an action or a "miss" indication, as described in
-Section [#sec-mau-semantics].
```

Comment:
The tables all implement the same data-plane interface ITable<TKey, TAction>, where TKey is a tuple. The data-plane semantics of a table is given solely by this interface, period. The interface, you will notice, does not depend on the match_kind. The implementation does.

Maybe the control-plane semantics depends on the match_kind, but the data-plane semantics does not.